### PR TITLE
update nix flake and add gofumpt

### PR DESCRIPTION
**Pull request by 6543**
just nix flake maintinance

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

File: flake.nix [hunk 1]
```diff
@@ -8,7 +8,7 @@
     flake-utils.lib.eachDefaultSystem (
       system:
       let
-        pkgs = nixpkgs.legacyPackages.${system};
+        pkgs = import nixpkgs { inherit system; };
```
Comment (jolheiser) [inline flake.nix]: I would prefer the `nixpkgs.legacyPackages.${system}` style. If anyone ever uses our flake in their own it won't cause an unnecessary evaluation.

**Review by jolheiser — APPROVED** [review]

**Comment by techknowlogick** [comment]
Why add gofmpt? I thought we run it similarly to golangci-lint

**Comment by jolheiser** [comment]
> Why add gofmpt? I thought we run it similarly to golangci-lint

Hm, true. This does have the benefit of having it in path for IDEs, though. I'm fine with it either way personally, I think

**Comment by 6543** [comment]
because then the tooling does not need install it and i can easy invoke it to format my code

**Comment by silverwind** [comment]
But can you lock down its version that way? New versions might introduce formatting differences.

**Comment by 6543** [comment]
well it should not but in this case we probably do want a version bumb anyway ... 

I would propose to let it stay untill we do notice a problem - and as i now actively use it more or less every day I would hit that fast

**Comment by silverwind** [comment]
I guess it'll be fine but if there is a way to version-lock at least some of these dependencies in this file to a certain version range, that'd be great.

**Pull request merged**